src/tools/mmaps_generator/PathGenerator.cpp

```diff
@@ -244,7 +243,8 @@ int main(int argc, char** argv)
 {
     Trinity::Banner::Show("MMAP generator", [](char const* text) { printf("%s\n", text); }, nullptr);
 
-    int threads = 3, mapnum = -1;
+    unsigned int threads = std::thread::hardware_concurrency();
```

Comment:
Hmm, not sure it makes it better. And 0 is a valid thread count to the program. The thread number is actually the number of threads additional to the program main thread.

Preventing 0 as thread count will disable one "feature" of the program as it is.
See this:
https://github.com/Rochet2/TrinityCore/blob/f41c26761df4272fe8343fc61439f7cc06af04c9/src/tools/mmaps_generator/MapBuilder.cpp#L212